server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageContentExtractor.java

```diff
@@ -89,8 +91,12 @@ private MessageContent parseFirstFoundMultipart(Multipart multipart) throws IOEx
             .orElse(MessageContent.empty());
     }
 
-    private String asString(TextBody textBody) throws IOException {
-        return IOUtils.toString(textBody.getInputStream(), textBody.getMimeCharset());
+    private Optional<String> asString(TextBody textBody) throws IOException {
+        String textBodyContent = IOUtils.toString(textBody.getInputStream(), textBody.getMimeCharset());
+        if (StringUtils.isEmpty(textBodyContent)) {
```

Comment:
We has asString(...) which return the optional of htmlBody or textBody
The problem here is, if we don't checking for empty string ("") then it always returns Optional.of(""). Because of StringBuilder
Then later, it should be difficult to check the empty html body or html text because it fact, it not empty, it is Optional.of(""). How can I deal with that?